test/benchmarks/benchmark_epsilon_decay.cc

```diff
@@ -0,0 +1,57 @@
+#include "../../vowpalwabbit/core/tests/simulator.h"
+#include "benchmarks_common.h"
+#include "vw/cache_parser/parse_example_cache.h"
+#include "vw/config/options_cli.h"
+#include "vw/core/learner.h"
+#include "vw/core/metric_sink.h"
+#include "vw/core/parser.h"
+#include "vw/core/reductions/epsilon_decay.h"
+#include "vw/core/setup_base.h"
+#include "vw/core/vw.h"
+#include "vw/io/io_adapter.h"
+#include "vw/text_parser/parse_example_text.h"
+
+#include <benchmark/benchmark.h>
+
+template <class... ExtraArgs>
+static void bench_epsilon_decay(benchmark::State& state, ExtraArgs&&... extra_args)
+{
+  std::array<std::string, sizeof...(extra_args)> res = {extra_args...};
+  std::string use_decay = res[0];
+  std::string model_count = res[1];
+  std::string bit_size = res[2];
+  std::string tolerance = res[3];
+
+  using callback_map =
+      typename std::map<size_t, std::function<bool(simulator::cb_sim&, VW::workspace&, VW::multi_ex&)>>;
+  callback_map test_hooks;
+
+  for (auto _ : state)
+  {
+    const size_t num_iterations = 1000;
+    const size_t seed = 99;
+    const std::vector<uint64_t> swap_after = {500};
+    if (use_decay == "decay")
```

Comment:
is this variable used like a boolean? if so maybe it should be of type bool.

Reply:
sure just did this get all the args in a consistent way, updated to bool